toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -809,3 +793,50 @@ func (r *RpmRepoCloner) reposArgsHaveOnlyLocalSources(reposArgs []string) bool {
 
 	return true
 }
+
+func tdnfDownload(args ...string) (err error, retriable bool) {
+	const (
+		unresolvedOutputPrefix = "No package"
+		unresolvedOutputSuffix = "available"
+	)
+
+	stdout, stderr, err := shell.Execute("tdnf", args...)
+
+	logger.Log.Debugf("stdout: %s", stdout)
+	logger.Log.Debugf("stderr: %s", stderr)
+
+	// ============== TDNF SPECIFIC IMPLEMENTATION ==============
+	//
+	// Check if TDNF could not resolve a given package. If TDNF does not find a requested package,
```

Comment:
Are we reasonably confident that, if tdnf prints "No package ... available", then retry is never useful? That sounds reasonable, but want to confirm based on what we've seen in the past.

Reply:
I think "reasonably confident" is a good way to describe the way I see it. Failures with the 502 error don't make TDNF assume the package is unavailable (the "No package... available" message is not printed out anywhere). I think it is "reasonable" to assume then, that other server or network errors won't cause this message to appear and we can thus trust that if TDNF says the package is not there, there's no point to retry.